### ISSUE TITLE
Add Module: Better Escape

### ISSUE THREAD

**Issue by Sduby22** [echasnovski/mini.nvim#86]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)

### Module(s)

https://github.com/max397574/better-escape.nvim

### Description

a better escaping module for those who prefer 'jk' or 'jj' to normal esc or Ctrl-C

This module provides better experience than simply `imap jk <esc>`, such as:

- Customizable delay which is independent from `timeoutlen`
- `j` will be immediately echoed back on screen when pressed. (if set `imap jk <esc>`, `j` wont be echoed until `timeoutlen` expired or you pressed something other than `k`. )

See: [Better Escape](https://github.com/max397574/better-escape.nvim)

**Comment by echasnovski** [comment]
That is an interesting suggestion. Thanks!

I don't think just mimicking this (already Lua based) plugin is the way to go, though. This functionality might be a part of some more general module that unobtrusively tracks keys and does something when certain key sequence is seen. I'll think about it.

**Comment by matu3ba** [comment]
Better alternatives are 1. https://github.com/anuvyklack/hydra.nvim, 2. https://github.com/Iron-E/nvim-libmodal and both dont rely on hacky things like `timeoutlen`.

**Comment by echasnovski** [comment]
> Better alternatives are 1. https://github.com/anuvyklack/hydra.nvim, 2. https://github.com/Iron-E/nvim-libmodal and both dont rely on hacky things like `timeoutlen`.

Yeah, I'll look more closely at them when attempting ['mini.clue'](https://github.com/echasnovski/mini.nvim#planned-modules).

**Comment by echasnovski** [comment]
This is now possible with [`MiniKeymap.map_combo()`](https://github.com/echasnovski/mini.nvim/blob/ee23e1abc206efc6d6cce19ec8c0a3da7a897035/doc/mini-keymap.txt#L370). Here are [examples of how to achieve "better escape" like behavior](https://github.com/echasnovski/mini.nvim/blob/ee23e1abc206efc6d6cce19ec8c0a3da7a897035/doc/mini-keymap.txt#L136-L151).

For any questions/recommendations use #1780.
